src/core/orderbookutils.js

```diff
@@ -50,7 +85,10 @@ OrderBookUtils.getOfferTakerPaysFunded = function(offer) {
 OrderBookUtils.getOfferTakerGets = function(offer) {
   assert(typeof offer, 'object', 'Offer is invalid');
 
-  return Amount.from_json(offer.TakerGets + IOU_SUFFIX);
+  let currency = offer.TakerPays.currency;
+  let issuer = offer.TakerPays.issuer;
+
+  return Amount.from_json(createAmount(offer.TakerGets, currency, issuer));
```

Comment:
Also see here:  https://github.com/ripple/ripple-lib/blob/5b4deabd90e5237bd0165e65520f25456d32a410/src/core/autobridgecalculator.js#L13

The code is used for `leg one` where TakerGets is supposed to be native, which makes sense I guess, where you want to trade some IOU for XRP, to then trade that XRP for the wanted IOU.

I guess that function needs some renaming / comments to make it clearer.